### DISCUSSION TITLE
2021-11-23 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#840] [Maintainer discussions]
## About
Each week (usually on Tuesday), we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Progress & Weekly Goals](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1690610)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1690620)
- [Upcoming State of Mathesar meeting](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1690626)
- [Seeking Team Event volunteers](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1690639)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1691071)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1693491)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1697037)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1697118)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1698049)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1699030)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/840#discussioncomment-1702920)

## Weekly check-in
@dmos62 @ghislaineguerin @mathemancer @pavish @seancolsen @silentninja please respond to the weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

- How did the last week go? What went well, what could have gone better?
- What are you planning to work on over the next week? Do you anticipate any blockers?
- I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## Progress & Weekly Goals

We've only had four days since the last discussion and it's a short week for some of us, so we're still going to be working on the some of the same goals as last week. Please comment below if you have any concerns or additions to make.

### Design
- @ghislaineguerin: Complete specs and finalize the design for #780 
- @ghislaineguerin: Complete specs and finalize the design for foreign keys:  https://github.com/centerofci/mathesar-wiki/pull/12
- @ppii775: Have a first draft of the viewing views design in the GitHub discussion stage: #466 
- @ghislaineguerin: Start design for #455

### Frontend
- Get 2-3 issues in **[[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)** at least to the draft PR stage. @seancolsen and @pavish, I'll leave it to you to figure how to divide this up, let me know if I can help.

### Backend
- Make progress on **[[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)**
  - @mathemancer: Get grouping refactor merged.
  - @dmos62: Get filtering for all unblocked data types done and merged.
  - @silentninja: Get column API and all unblocked display options done and merged.
  - @silentninja and @eito-fis: Get `TIME` and `TIMESTAMP` types done.
- @kgodey: Spec for API error structure: #560 

### Product
- @kgodey: Finish specs and review for remaining Views issues and get them ready for design work: #442

**Reply by seancolsen** [reply]
> Get all issues in [07] Working with Tables at least to the draft PR stage

@kgodey by my count, that's 10 tickets that need new PRs. Are you requesting draft PRs for all 10 within the next week? That would be much faster than I've observed us to work thus far.

**Reply by kgodey** [reply]
@seancolsen Some of them seem small enough that I was hoping they would be fairly quick e.g.
- #447 
- #497 
- #631 

If that's too ambitious, what do you think would be a good goal?

**Reply by kgodey** [reply]
Actually, I looked through the list of issues again and I'm not sure what I was thinking, especially since this is Thanksgiving week. I updated the goals above to be 4-5 issues instead of "all issues".

**Reply by seancolsen** [reply]
#447 is unfortunately not quick. I added some [more thoughts](https://github.com/centerofci/mathesar/issues/447#issuecomment-977411896) within the ticket about how we could make it quick.

**Reply by pavish** [reply]
@kgodey Some of these issues look simple but they have a lot of prerequisites. eg.,
* #447 as mentioned by @seancolsen ,
* #497 would involve an UX discussion on the column header click action since it collides with the current column rename UX,
* #500 which I'm currently working on. This issue should have ideally been placed with the data-types milestone since the default value of a column depends on the type (it involves validation, showing the correct input element etc.,). I started working on it because I wanted to structure a common type configuration on the frontend and testing it with the usecase of default value seemed like a good idea. It's going be a big PR but it will make it easier to complete the data types milestone.

I don't think we'll be able to do draft 4 PRs this week. @seancolsen's out for half of it and he's also working on issues that are not on this milestone, and I need the entire week for #500. We would probably end up with only 2 or 3.

**Reply by kgodey** [reply]
@pavish Thanks. I've updated the goal to 2-3. If UX discussions are necessary to resolve some issues, I recommend starting those soon since they'll require some turnaround time.

**Comment by kgodey** [comment]
## Comms Assignee schedule

FYI, we may adjust the December schedule closer to the holidays once we figure out when people are taking time off.

### Schedule

| Week | Person |
|-|-|
| Nov 22 to Nov 26 | Kriti |
| Nov 29 to Dec 3 | Pavish |
| Dec 6 to Dec 10 | Mukesh |
| Dec 13 to Dec 17 | Sean |
| Dec 20 to Dec 24 | Dominykas |
| Dec 27 to Dec 31 | Ghislaine |
| Jan 3 to Jan 7 | Brent |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by kgodey** [comment]
## Upcoming State of Mathesar meeting
Now that we've completed hiring, I'll be scheduling a "State of Mathesar" meeting for the core team where we'll go over where we are and our goals for the next few months. It'll probably be next week.

Please suggest any agenda items you think would be useful to cover. Here's what I have so far:
- Alpha release timeline and goals
- Long term vision for Mathesar
- Quick walkthrough of team processes and workflow

**Comment by kgodey** [comment]
## Seeking Team Event volunteers

Anyone interested in organizing our next core team event (or two or three?)

Here's our [Team Event wiki page](https://wiki.mathesar.org/team/guide/events) for inspiration.

**Reply by seancolsen** [reply]
I'll take the next one

**Reply by kgodey** [reply]
Thanks @seancolsen!

**Comment by seancolsen** [comment]
## Sean's check in

How did the last week go? What went well, what could have gone better?

- Going well. Nothing stands out as particularly remarkable.

What are you planning to work on over the next week? Do you anticipate any blockers?

- Table re-naming
- Next week will be short since I'm not working Wednesday, Thursday, or Friday.

I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?

- Cool. Any day works for me.

Anything else you'd like to add?

- Nope

**Comment by dmos62** [comment]
## Dominykas' check in

How did the last week go? What went well, what could have gone better?

- Going well. Took some time to figure out how to model filtering. I've enjoyed informal calls with people on the team.

What are you planning to work on over the next week? Do you anticipate any blockers?

- I hope I'll have the [number filtering PR](https://github.com/centerofci/mathesar/pull/837) fleshed out; though I expect to have to do some finishing touches next week;
- Might continue working on filtering for other types;
- These tasks have also been suggested:
  - https://github.com/centerofci/mathesar/issues/592;
  - https://github.com/centerofci/mathesar/issues/280.

I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?

- I've too little experience with it for my input to have weight, but off the top of my head it's nice to finish a week with a review/discussion, because it gives the week a sense of completion and the weekend a sense of rest because you know what's you'll do next week.

Anything else you'd like to add?

- No

**Comment by ghislaineguerin** [comment]
## Ghislaine's Check-In 
> How did the last week go? What went well, what could have gone better?

It was regular. I didn't work a whole week, so my estimations on how much work I could do were a bit off. I also wish I could reply to other team discussions more promptly.

> What are you planning to work on over the next week? Do you anticipate any blockers?

I want to start with Views and close the Foreign Key spec. There are a couple of components that need to be defined that weren't part of the original spec. I might consider creating a separate issue to define specs for constraints settings.

> I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?

I think that makes sense. 

> Anything else you'd like to add?

No

**Comment by kgodey** [comment]
## Kriti's check in

> How did the last week go? What went well, what could have gone better?

Pretty well. I think we have most of Views nailed down now, which is nice.

> What are you planning to work on over the next week? Do you anticipate any blockers?

- Get remaining Views issues ready for work.
  - I think I'm going to move #806 to a later milestone, probably [Computed Data](https://github.com/centerofci/mathesar/milestone/12) since it's not directly related to basic support for Views. 
- #560 
- Write wiki pages describing the data type concept and get feedback, should help with resolving:
  - #838 
  - #843
- Board meeting presentation prep
- Investigate staging server deployment failure

**Comment by mathemancer** [comment]
> How did the last week go? What went well, what could have gone better?

Things are progressing.  The grouping API work is coming along, and I think we've also done well getting some design issues hashed out.  I also enjoyed the team event.  I think we can do better at organizing discussions around different topics.  At the moment, we have a number of different channels both on Matrix and Github, and figuring out the source of truth for some decision or outcome is becoming a bit tricky.

> What are you planning to work on over the next week? Do you anticipate any blockers?

My top priority is the Grouping API, and getting it merged.  After that, I'm going to pick up on other type work if needed.  The only blockers I foresee are due to the fact that Thanksgiving holidays will increase turnaround on PR reviews.

> I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?

No disagreement from me; the Tuesday thing isn't working out as well as I'd hoped.  Maybe that's an idea that doesn't work properly asynchronously.  In fact, if it's not too much burden for @kgodey , it might make sense to even have that prepared before or during the weekend, so that those of us in earlier time zones can get to it first thing before we get into working on other stuff for the week.

> Anything else you'd like to add?

I think that we (especially I) could use the "concepts" section of the wiki more and better.  We have a huge amount of terminology, and that's the only real source of truth for what we're calling different concepts, how they interrelate, and so on.  Unfortunately, it remains a bit sparse.  I'll try to make a personal effort to update that section with outcomes of discussions around how to define a particular concept in the future.

I was also thinking of updating the wiki with kind of a "links section" dedicated mostly to links to the most relevant bits of the PostgreSQL (and maybe SQLAlchemy) docs.  I keep having to find the same page describing the Lexical Structure, for example.

**Reply by kgodey** [reply]
I'm going to be working on fleshing out the "Concepts" section as well.

I think the idea of a Links section sounds great. Perhaps a subsection or sub-page of the Engineering [Resources](https://wiki.mathesar.org/engineering/resources) page?

**Comment by pavish** [comment]
## Pavish's check in

> How did the last week go? What went well, what could have gone better?

The week went well. PR reviews and discussions aside, I spent some time analyzing types configuration for display, selection, validation and input. 

> What are you planning to work on over the next week? Do you anticipate any blockers?

This stays the same as last week. I'm planning to complete Setting default value for column PR, which includes the above mentioned type configuration. I'll then work on column renaming if I get the time.

> I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?

I think that's great. It might be better if the weekly discussion is posted on Friday @kgodey 's timezone, so that others in earlier timezones can get to it on their Monday.

> Anything else you'd like to add?

Nothing at the moment.

**Comment by silentninja** [comment]
## Mukesh's check in

> How did the last week go? What went well, what could have gone better?

The week went well. Was able to make the necessary review changes for the #815 Django Column model PR and add few enhancements to #822 Display option PR. 

> What are you planning to work on over the next week?

Will be working on #424 Timestamp type

> Do you anticipate any blockers?

None

> I'm thinking about moving the weekly discussion to Mondays to kick off the week. Any thoughts on that?

I would gladly welcome this. It would be better if it can be posted end of the week, so we can start off the next week with the check in

> Anything else you'd like to add?

Nothing at the moment.
